src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.PackTool.targets

```diff
@@ -80,14 +88,26 @@ Copyright (c) .NET Foundation. All rights reserved.
        ToolsSettingsFilePath="$(_ToolsSettingsFilePath)" />
   </Target>
 
-  <Target Name="ResolveApphostAsset" DependsOnTargets="ResolveReferences;Publish">
+  <!--
+    ============================================================
+                                        GenerateShimsAssets
+
+    Generate packaged shims for signing when PackAsToolShimRuntimeIdentifiers is set
+    ============================================================
+    -->
+
+  <Target Name="GenerateShimsAssets"
+          AfterTargets="Build"
+          DependsOnTargets="ResolvePackageAssets"
+          Condition="'$(PackAsTool)' == 'true'" >
+
     <ItemGroup>
       <_PackAsToolShimRuntimeIdentifiers Include="$(PackAsToolShimRuntimeIdentifiers)"/>
     </ItemGroup>
 
     <GenerateShims
       DotNetAppHostExecutableNameWithoutExtension="$(_DotNetAppHostExecutableNameWithoutExtension)"
-      PackagedShimOutputDirectory="$(IntermediateOutputPath)/$(Configuration)/shims/$(TargetFramework)"
+      PackagedShimOutputDirectory="$(OutDir)/shims/$(TargetFramework)"
```

Comment:
@nguerrera and just change IntermediateOutputPath to OutDir (/obj to /bin)is good enough as we discussed right

Reply:
Nit for the smaller change before this one to just let this be customized, we actually don't need $(Configuration) because the IntermediateOutputPath is configuration-specific (unlike BaseIntermediateOutputPath), default it obj\$(Configuration), so here we're making obj\Debug\Debug\shims.